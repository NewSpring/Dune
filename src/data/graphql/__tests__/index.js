import { typeDefs, resolvers, graphql } from "../";

describe("typeDefs", () => {
  it("should match a snapshot", () => {
    expect(typeDefs).toMatchSnapshot();
  });
});

describe("resolvers", () => {
  const { Query, Response, Page, Block } = resolvers;

  describe("Query", () => {
    const { sample } = Query;
    it("has an expected shape", () => {
      expect(sample()).toEqual({ code: 200, message: "hello world" });
    });
  });

  describe("getRouteInfo", () => {
    const { getRouteInfo } = Query;

    it("has resolve", () => {
      expect(getRouteInfo).toBeDefined();
    });

    it("returns a sample registry if path is root", () => {
      expect(getRouteInfo(null, { path: "/" })).toEqual({
        id: 1,
        layout: "Horizontal",
        blocks: [
          { id: 3, order: 1, zone: "main", path: "HelloWorld" },
          { id: 2, order: 1, zone: "secondary", path: "HelloWorld" },
          { id: 1, order: 0, zone: "secondary", path: "Counter" },
        ],
      });
    });
  });

  describe("Response", () => {
    const data = Query.sample();
    const { code, message } = Response;
    it("has a code method", () => {
      expect(code(data)).toEqual(200);
    });
    it("has a message method", () => {
      expect(message(data)).toEqual("hello world");
    });
  });

  describe("Page", () => {
    const { id, blocks, layout } = Page;
    it("has an id method", () => {
      expect(id({ id: 5 })).toEqual(5);
    });
    it("has a layout method", () => {
      expect(layout({ layout: "foo" })).toEqual("foo");
    });
    it("has a blocks method", () => {
      expect(blocks({ blocks: [{ id: 1 }, { id: 2 }] })).toEqual([
        { id: 1 },
        { id: 2 },
      ]);
    });
  });

  describe("Block", () => {
    const { id, path, zone, order } = Block;
    it("has an id method", () => {
      expect(id({ id: 1 })).toEqual(1);
    });
    it("has a zone method", () => {
      expect(zone({ zone: "main" })).toEqual("main");
    });
    it("has an order method", () => {
      expect(order({ order: 1 })).toEqual(1);
    });
    it("has a path method", () => {
      expect(path({ path: "Hello" })).toEqual("Hello");
    });
  });
});

describe("graphql", () => {
  it("has a sample root level query field", async () => {
    const { data } = await graphql("{ sample { code message } }");
    expect(data).toEqual({ sample: { code: 200, message: "hello world" } });
  });
});

describe("getRouteInfo", () => {
  it("exists", () => {});
});
