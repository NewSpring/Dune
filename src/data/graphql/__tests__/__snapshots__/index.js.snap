// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeDefs should match a snapshot 1`] = `
"
type Response {
  code: Int!
  message: String
}

type Page {
  id: Int!
  layout: String!
  blocks: [Block]
}

type Block {
  id: Int!
  path: String!
  zone: String!,
  order: Int!,
}

# the schema allows the following query:
type Query {
  sample: Response
  getRouteInfo(path: String!): Page
}

schema {
  query: Query
}
"
`;
